src/collective/ifttt/profiles/default/actions.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<object name="portal_actions" meta_type="Plone Actions Tool"
+   xmlns:i18n="http://xml.zope.org/namespaces/i18n">
+ <object name="object" meta_type="CMF Action Category">
+  <object name="Add IFTTT Content Trigger" meta_type="CMF Action" i18n:domain="plone">
+   <property name="title" i18n:translate="">Add IFTTT Content Trigger</property>
+   <property name="description" i18n:translate="">Add new IFTTT Content Trigger</property>
+   <property name="url_expr">string:$object_url/add_ifttt_trigger</property>
+   <property name="icon_expr"></property>
+   <property name="available_expr">folder/cb_dataValid|nothing</property>
```

Comment:
Instead cb_dataValid you probably want a custom browser view, which returns True if IFTTT plugin is properly configured and rule can be created.

Reply:
https://training.plone.org/5/mastering-plone/views_2.html

Reply:
Also, how to check if the rule can be created?

Reply:
For me it seems that we could reuse code in plone.app.contentrules with condition (I have not tested that myself):

```
<property name="available_expr">folder/plone_contentrules_info/show_rules_tab|nothing</property>
```

That should call a method `show_rules_tab` of view `plone_contentrules_info` which is defined and registered as follows

https://github.com/plone/plone.app.contentrules/blob/afe96d4c76bbd71a72adcac3552cb9a55a15429b/plone/app/contentrules/browser/info.py

https://github.com/plone/plone.app.contentrules/blob/afe96d4c76bbd71a72adcac3552cb9a55a15429b/plone/app/contentrules/browser/configure.zcml

Reply:
Actually, you need to copy that view into your code and register it with different name (eg. ifttt_contentrules_info). Then you need to remove the following lines

```python
        if not storage:
            return False
```

That code actually checks if there are any rules available and hides the content rules menu when there is no rules created yet. But in our use case, it should be able to use IFTTT action menu item to create a rule even when the site has no rules yet.